lib/data/models/activity_model.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:isar/isar.dart';
+import 'package:movna/domain/entities/sport.dart';
+
+part 'activity_model.g.dart';
+
+@Collection()
+class ActivityModel {
+  /// Use startTime as unique identifier.
+  Id get id => startTime.microsecondsSinceEpoch;
+
+  DateTime startTime = DateTime.fromMillisecondsSinceEpoch(0);
+
+  DateTime? stopTime;
+
+  // TODO : add field timezoneOffset or timezone to store localTime as well,
+  // In order to be able to restore the original date times, which were
+  // converted to UTC when stored in database.
+
+  @Enumerated(EnumType.name)
+  Sport? sport;
+
+  String? name;
+
+  double distanceInMeters = 0;
+
+  int durationInMicroSeconds = 0;
+
+  double maxSpeedInMetersPerSecond = 0;
+
+  double averageSpeedInMetersPerSecond = 0;
+
+  double? averageHeartBeatPerMinute;
+
+  String notes = '';
+
+  List<TrackSegmentModel> trackSegments = [];
+}
+
+@embedded
+class TrackSegmentModel {
+  List<TrackPointModel> trackPoints = [];
+}
+
+@embedded
+class TrackPointModel {
+  LocationModel? location;
```

Comment:
why would the location be null ?

Reply:
I the user decides to track an activity without enabling the GPS, we should be able to store timestamps and heart rates (and other in the future). But the current implementation prevents that since the timestamp is in the `Location` class. I do not like that at all, but I originally did that to copy the implementation of geolocator's position.
`Location` may need to be broken down to `Location` with all position-related fields, and `TimedLocation`, that inherits `Location` or has a `Location` field. This way we could use this modified `Location?` class in `TrackPoint`, and still get `TimedLocation` from the location repository.

Yes that does not answer your question, the original goal was that : to be able not to have GPS data in `TrackPoint`, but as of now doing this prevents from having timestamps.

Reply:
Ok so the location can be null depending on the goal of the app
If it's supposed to be an app primarily used to track your location during a physical activity and eventually some other stats to follow along your route I think location should not be null.
If it's supposed to be an app tracking physical activity and location is just a type of data among others then it can be null

In my mind it was designed as the first option, and eventually you could see other data such as heartrate if available.
What I anticipate is that we'll need to perform operations on lists of trackpoints and I feel that enabling `null` location will be troublesome when querying the database, and for a use-case (follow your activity without any speed, or position information) that is not really a need.

Basically I just want to make sure that we are not being to generic and if maybe we could just assume / force some major usages

Reply:
I prefer to let things as they are, this enables to use the app without any GPS data. The user could for instance track its activity just using its duration. They would only know that that day they practiced this sport for this duration, and I think this is an important feature if their device does not have a GPS or if geolocalisation has been disabled for battery optimization. They could then manually provide the total ran distance if they want to.